### PR TITLE
feat(deadline): allow providing subnets for RenderQueue's ALB

### LINKED ISSUE #263: Allow subnet selection for RenderQueue's ALB
The ask is for the ability to control which subnets the RenderQueue will deploy its ALB(s) into.

### Use Case

When deploying into a Local Zone (ex: the LA Local Zone), the Local Zone is actually a part of a greater region (us-west-2 in the case of the LA zone). An ALB cannot be deployed into subnets that are both in the region and in the local zone. So, to support local zones we need the ability for the user to select which subnets the ALB will be created within (either entirely in the LZ or entirely in the regular Region AZs).

### Proposed Solution

Adding a `vpcAlbSubnets` property to the RenderQueue props.

### Other

N/A

* [X] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by ddneilson**
Implements: https://github.com/aws/aws-rfdk/issues/263

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-rfdk/lib/deadline/lib/render-queue.ts [hunk 1]
```diff
@@ -334,6 +334,7 @@ export class RenderQueue extends RenderQueueBase implements IGrantable {
 
     const loadBalancer = new ApplicationLoadBalancer(this, 'LB', {
       vpc: this.cluster.vpc,
+      vpcSubnets: props.vpcAlbSubnets ?? { subnetType: SubnetType.PRIVATE },
```
Comment (horsmand) [inline packages/aws-rfdk/lib/deadline/lib/render-queue.ts]: Do we need to add the `?? { subnetType: SubnetType.PRIVATE }` part here? This matches the default behaviour, but why not let the constructor handle this defaulting itself?

Comment (ddneilson) [reply packages/aws-rfdk/lib/deadline/lib/render-queue.ts]: Don't *need* it. I just like being explicit.

**Review by horsmand — DISMISSED** [review]

**Review by horsmand — DISMISSED** [review]

File: packages/aws-rfdk/lib/deadline/lib/render-queue-ref.ts [hunk 2]
```diff
@@ -243,6 +243,13 @@ export interface RenderQueueProps {
    */
   readonly vpcSubnets?: SubnetSelection;
 
+  /**
+   * The subnets into which to place the Application Load Balancer that is deployed.
+   *
+   * @default - One Private subnet from each AZ.
+   */
+  readonly vpcAlbSubnets?: SubnetSelection;
```
Comment (jusiskin) [inline packages/aws-rfdk/lib/deadline/lib/render-queue-ref.ts]: nit: does it make more sense to name this `vpcSubnetsAlb`? Unfortunately I had created the original `vpcSubnets` which dictates the placement of the ASG instances without thinking about the ALB. The original `vpcSubnets` came from the convention used in the CDK's standard AWS construct library:

* [`Instance` construct's `vpcSubnets` prop](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-ec2.Instance.html#vpcsubnets)
* [`AutoScalingGroup` construct's `vpcSubnets` prop](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-autoscaling.AutoScalingGroup.html#vpcsubnets)

To continue with that convention as best as possible, I suggest we stick with `vpcSubnets` prefix and append `Alb` as a suffix. That way those using code-completion will see both show up and can pick/explore accordingly. What do you think though?

**Review by jusiskin — CHANGES_REQUESTED** [review]
Just a minor prop rename suggestion - otherwise looks good

File: packages/aws-rfdk/lib/deadline/lib/render-queue-ref.ts (hunk 2 above)
Comment (ddneilson) [reply packages/aws-rfdk/lib/deadline/lib/render-queue-ref.ts]: That seems reasonable to me.

**Review by horsmand — APPROVED** [review]

File: packages/aws-rfdk/lib/deadline/lib/render-queue-ref.ts (hunk 2 above)
Comment (jusiskin) [reply packages/aws-rfdk/lib/deadline/lib/render-queue-ref.ts]: Looks good - thanks

**Review by jusiskin — APPROVED** [review]

**Pull request merged**